pkg/sources/adapter/salesforcesource/client/bayeux_test.go

```diff
@@ -0,0 +1,350 @@
+/*
+Copyright 2021 TriggerMesh Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"github.com/triggermesh/triggermesh/pkg/sources/adapter/salesforcesource/auth"
+	"github.com/triggermesh/triggermesh/pkg/sources/adapter/salesforcesource/auth/fake"
+	zapt "go.uber.org/zap/zaptest"
+)
+
+const (
+	tAPIVersion      = "53.0"
+	tClientID        = "abcde"
+	tResponseChannel = "/test/channel"
+	tMetaChannel     = "/meta"
+	tReconnect       = "handshake"
+	// tResponseFinishChannel will signal that the
+	// test case has completed and the server should
+	// not expect new events.
+	tResponseFinishChannel = "/test/finish"
+)
+
+var (
+	tSubscription = []Subscription{
+		{
+			Channel:  "/channel1",
+			ReplayID: -1,
+		},
+	}
+)
+
+type response struct {
+	handshake []HandshakeResponse
+	subscribe []SubscriptionResponse
+	connect   []ConnectResponse
+}
+
+func TestBayeux(t *testing.T) {
+
+	logger := zapt.NewLogger(t).Sugar()
+
+	testCases := map[string]struct {
+		headers map[string]string
+
+		// responses need follow the order they are
+		// expected to be requested.
+		responses []response
```

Comment:
This responses element is partly a parameter for the test case, and partly an expectation for requests received from the Bayeux polling loop.